browser/component_updater/brave_component_updater_delegate.cc

```diff
@@ -27,6 +27,7 @@ BraveComponentUpdaterDelegate::BraveComponentUpdaterDelegate()
 BraveComponentUpdaterDelegate::~BraveComponentUpdaterDelegate() = default;
 
 void BraveComponentUpdaterDelegate::Register(
+    const std::string& component_id,
```

Comment:
What are these changes needed for?

Reply:
All of our component installer policies require this in the initializer. I based my general purpose one on the existing ones and as a result I need this variable.

Reply:
How did this function in desktop fine without it then?

Reply:
In iOS we need it to get the install directory (see component installer policy below)
```
return base::FilePath::FromUTF8Unsafe(component_id_);
```
This is what we do for all of our installer policies we have. You can see we do this in a number of places: https://github.com/search?q=repo%3Abrave%2Fbrave-core+return+base%3A%3AFilePath%3A%3AFromUTF8Unsafe%28component_id_%29%3B&type=code
I wanted to stick to something I know works in iOS but to do that I need this component id.

In desktop they do this: 
https://github.com/brave/brave-core/blob/2dd03b6f9818dc8d365fefd28f964456ea341ccc/browser/component_updater/brave_component_installer.cc#L130

Though I don't fully understand what is going on there, its clear that the component id is not needed.

Reply:
It says in a comment right above that they are driving the component id from the public kye, thus not needing to pass in the component id

Reply:
Ok removed the `component_id` as requested. 

But I still find it odd that the `Register` function does not mirror the `Unregister` function which uses the passed `component_id` instead of deriving it.